src/relax/transform/lift_transform_params.cc

```diff
@@ -494,22 +700,71 @@ class ConsumeBundledParams : public ExprMutator {
   std::unordered_map<int, Expr> param_remap_;
 };
 
+std::vector<std::pair<GlobalVar, Function>> GetTargetFunctions(
+    const IRModule& mod, const Array<String>& target_function_names) {
+  std::vector<std::pair<GlobalVar, Function>> target_functions;
+  if (target_function_names.size()) {
+    for (const auto& name : target_function_names) {
+      auto gvar = mod->GetGlobalVar(name);
+      target_functions.push_back({gvar, Downcast<Function>(mod->Lookup(gvar))});
+    }
+  } else {
+    // Get all the functions that have the `num_input` attribute.
+    for (const auto& [gvar, func] : mod->functions) {
+      if (func->IsInstance<FunctionNode>()) {
+        auto opt_num_input = func->GetAttr<Integer>(attr::kNumInput);
+        if (opt_num_input) {
+          target_functions.emplace_back(gvar, Downcast<Function>(func));
+        }
+      }
+    }
+    std::sort(target_functions.begin(), target_functions.end(),
+              [](const auto& lhs, const auto& rhs) {
+                return lhs.first->name_hint < rhs.first->name_hint;
+              });
+  }
+  return target_functions;
+}
+
 }  // namespace
 
 namespace transform {
 
-Pass PartitionTransformParams() {
+Pass PartitionTransformParams(bool shared_transform, const Array<String>& target_function_names) {
   auto pass_func = [=](IRModule mod, PassContext pc) {
-    PreprocessPartitioner mutator;
-
     IRModule updates;
-    for (const auto& [gvar, func] : mod->functions) {
-      if (auto opt = func.as<relax::Function>()) {
-        auto new_func = Downcast<Function>(mutator(opt.value()));
-        if (!new_func.same_as(func)) {
-          updates->Add(gvar, new_func);
-        }
+
```

Comment:
With accepting a `Variant<Bool, Array<String>>` for the `shared_transform` argument, we would then check whether we need to make the shared transform as follows:

```c++
bool generate_shared_transform = shared_transform.as<Bool>().value_or(Bool(true))->value;
```